File: jFlash/Classes/AppSpecific/JapaneseCard.m
//
//  JapaneseCard.m
//  jFlash
//
//  Created by Mark Makdad on 8/21/11.
//  Copyright 2011 Long Weekend LLC. All rights reserved.
//

#import "JapaneseCard.h"
#import "ExampleSentencePeer.h"

@implementation JapaneseCard

@synthesize romaji;

- (void) dealloc
{
  [romaji release];
  [super dealloc];
}

- (void) hydrateWithResultSet:(FMResultSet*)rs
{
  [self hydrateWithResultSet:rs simpleHydrate:NO];
  self.romaji = [rs stringForColumn:@"romaji"];
	self._headword    = [rs stringForColumn:@"headword"];
}

- (void) hydrateWithResultSet:(FMResultSet*)rs simpleHydrate:(BOOL)isSimple
{
  [super hydrateWithResultSet:rs simpleHydrate:isSimple];
  self.romaji = [rs stringForColumn:@"romaji"];
	self._headword    = [rs stringForColumn:@"headword"];
}

/**
 * Returns YES if a card has example sentences attached to it
 * \param newVersion - If YES, uses 1.2 version Sql, if NO, uses 1.1 version
 */
- (BOOL) hasExampleSentencesWithPluginManager:(PluginManager *)pluginManager;
{
  BOOL returnVal = NO;
  if ([pluginManager pluginKeyIsLoaded:EXAMPLE_DB_KEY]) // we always have a sentence if the plugin is not installed
  {
    returnVal = [ExampleSentencePeer sentencesExistForCardId:self.cardId];
  }
  return returnVal;
}

/** depending on APP_READING value in settings, will return a combined reading */
- (NSString*) reading
{
  NSUserDefaults *settings = [NSUserDefaults standardUserDefaults];
  NSString *combined_reading = nil;
  
  // Mux the readings according to user preference
  if([[settings objectForKey:APP_READING] isEqualToString:SET_READING_KANA])
  {
    combined_reading = [NSString stringWithFormat:@"%@", [self hw_reading]];
  } 
  else if([[settings objectForKey:APP_READING] isEqualToString: SET_READING_ROMAJI])
  {
    combined_reading = [NSString stringWithFormat:@"%@", [self romaji]];
  }
  else
  {
    // Both together
    combined_reading = [NSString stringWithFormat:@"%@ - %@", [self hw_reading], [self romaji]];
  }
  
  return combined_reading;
}


@end

File: android/src/com/longweekendmobile/android/xflash/model/JapaneseCard.java
package com.longweekendmobile.android.xflash.model;

//  JapaneseCard.java
//  Xflash
//
//  Created by Todd Presson on 1/17/12.
//  Copyright 2012 Long Weekend LLC. All rights reserved.
//
//
//  public JapaneseCard()
//
//  public void hydrate(Cursor  )       @over
//  public void hydrate(Cursor  ,boolean  ) @over 
//
//  public boolean hasExampleSentencesWithPluginManager(Context  )
//  public String reading()
//
//  public void setRomaji(String  )
//  public String getRomaji()
//
//  + all methods inherited from Card

import android.database.Cursor;

import com.longweekendmobile.android.xflash.XflashSettings;

public class JapaneseCard extends Card
{
    // private static final String MYTAG = "XFlash JapaneseCard";

    private String romaji;

    public JapaneseCard()
    {

    }

    // override of Card.hydrate(Cursor  ) that includes JapaneseCard
    // relevant columns from the database
    @Override
    public void hydrate(Cursor inCursor)
    {
        // make a call up the line to Card.hydrate(Cursor,  boolean)
        super.hydrate(inCursor,false);

        int tempColumn = inCursor.getColumnIndex("romaji");
        romaji = inCursor.getString(tempColumn);

        tempColumn = inCursor.getColumnIndex("headword");
        headword = inCursor.getString(tempColumn);
    }


    // override of Card.hydrate(Cursor  ) that includes JapaneseCard
    // relevant columns from the database
    @Override
    public void hydrate(Cursor inCursor,boolean isSimple)
    {
        // make a call up the line to Card.hydrate(Cursor,  boolean)
        super.hydrate(inCursor,isSimple);
        
        int tempColumn = inCursor.getColumnIndex("romaji");
        romaji = inCursor.getString(tempColumn);

        tempColumn = inCursor.getColumnIndex("headword");
        headword = inCursor.getString(tempColumn);
    }

    // returns 'true' if a card has example sentences attached to it
    // TODO - don't have adequate info to properly replace PluginManager
    public boolean hasExampleSentencesWithPluginManager()
    {
        boolean returnVal = false;

        // we always have a sentence if the plugin is not installed
        // if ([pluginManager pluginKeyIsLoaded:EXAMPLE_DB_KEY])
/*      
        if( false )
        {
            returnVal = ExampleSentencePeer.sentencesExistForCardId(cardId);
        }
*/

        return returnVal;

    }  // end hasExampleSentencesWithPluginManager(  )


    // depending on APP_READING value in SharedPreferences, will
    // return a combined reading
    public String reading()
    {
        String combinedReading;

        int tempReadingMode = XflashSettings.getReadingMode();

        // mux the readings according to user preferences
        if( tempReadingMode == XflashSettings.LWE_READINGMODE_KANA )
        {
            combinedReading = hw_reading;
        }
        else if( tempReadingMode == XflashSettings.LWE_READINGMODE_ROMAJI )
        {
            combinedReading = romaji;
        }
        else
        {
            combinedReading = hw_reading + " - " + romaji;
        }

        return combinedReading;
    
    }  // end reading()
  
    // generic getters/setters
    public void setRomaji(String inRomaji)
    {
        romaji = inRomaji;
    }

    public String getRomaji()
    {
        return romaji;
    }

}  // end JapaneseCard class declaration




